### DISCUSSION TITLE
Concern with hydration data structure stability

### DISCUSSION THREAD

**Discussion by chrisvasz** [TanStack/query#1502] [Q&A] [answered]
First of all, loving v3!

I have a question about this line from the docs for [hydration/dehydrate](https://react-query.tanstack.com/reference/hydration/dehydrate):
> You should not rely on the exact format of this response, it is not part of the public API and can change at any time.

If the data structure can change at any time, does that mean you don't guarantee backwards compatibility? For example, if v3.5 changes the hydration data structure, will it be able to load the structure from 3.4? What if 3.6 changes the structure again? Will it be able to load data structures from both 3.4 and 3.5?

I'm asking because I want to use the builtin hydration/dehydration, but if you can't guarantee backwards compatibility, then I'll probably have to roll my own. We use react-query for our Cordova app and if the app fails to dehydrate, then our users could lose data – specifically any pending offline transactions they have accumulated by using the app without a data connection (among other things).

If you can't guarantee backwards compatibility for evolving hydration data structures, could you possibly communicate about the changes in the release notes? I know the structure isn't part of the public API, but it would still be really nice!

**Comment by boschni** [comment, marked as answer]
Hi @chrisvasz! Think this is a good point and we should definitely try to keep the structure backwards compatible as otherwise hydration might break between versions. Incompatible structure changes should be well documented and released in a major version.
